suite/tests/linux/clone.c

```diff
@@ -116,6 +119,40 @@ test_thread(bool share_sighand, bool clone_vm, bool use_clone3)
     delete_thread(child, stack);
 }
 
+#ifdef X86
+
+/* i#6514: Test passing NULL for the stack pointer to the syscall. */
+void
+test_with_null_stack_pointer(bool clone_vm, bool use_clone3)
+{
+    print("%s(clone_vm %d, use_clone3 %d)\n", __FUNCTION__, clone_vm, use_clone3);
+    int flags = clone_vm ? (CLONE_VFORK | CLONE_VM) : 0;
+    int ret;
+    /* If we don't have SYS_clone3, keep expected output the same and just use SYS_clone.
+     */
+    bool really_use_clone3 = use_clone3;
+#    ifndef SYS_clone3
```

Comment:
For core DR the presence of a SYS_ constant does not mean the syscall is supported by the underlying kernel: we have to support building on one machine and running on another.  For tests that's not nec the case, but here don't we get our SYS constants from the DR headers, which will define them no matter what the underlying kernel supports?  I believe this has to do a check for ENOSYS.  I think that's what #6596 is about, right?

Reply:
While you are correct that compile time presence of SYS_constant does not guarantee runtime presence of the syscall, the current patch is not trying to also solve this problem and instead sticks with what is already there.
Compare this code with the current definition of test_thread().

Reply:
The problem is, what is already there is breaking ongoing testing efforts and is being actively changed concurrently with this PR to fix that other testing: #6596.  So this new code should match the post-#6595-fixed code to avoid regressing and breaking the container testing yet again.

Reply:
I'm aware of #6596. One of these will go in first and the other can be updated then.
It's trivial to update either when that happens.

Reply:
> I'm aware of https://github.com/DynamoRIO/dynamorio/issues/6596.

Please make this clear by having a TODO comment so the reader knows this ifdef is not meant to be permanent.

> One of these will go in first and the other can be updated then.
> It's trivial to update either when that happens.

I don't know that it's that simple: my fear is that there is no automated container test in our GA CI tests, so if this PR goes in first and the author of the other just clicks on "merge" and the automated merge works, that author may have no idea that this PR has broken the container test.  It is not unusual to not re-do all manual testing after every merge from master.  So it seems there is a decent chance this PR would break what the other PR is trying to fix.  I would suggest putting a comment on the other PR so they are aware of this PR potentially breaking them.

Reply:
Fixes for #6596 have gone in and this patch has been updated with those changes.